.github/workflows/publish_release_candidate.yml

```diff
@@ -0,0 +1,67 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Publish Release Candidate
+
+on:
+  push:
+    tags:
+    # Rrigger workflow when a tag whose name matches the pattern
+    # "apache-arrow-{MAJOR}.{MINOR}.{PATCH}-rc{RC_NUM}" is pushed.
+      - "apache-arrow-*-rc*"
+
+jobs:
+  publish-release-candidate:
+    name: Publish Release Candidate to GitHub Releases
+    runs-on: ubuntu-latest
+    if: github.ref_type == 'tag'
+    steps:
+      - name: Checkout apache/arrow
+        uses: actions/checkout@v4
+        with:
+          path: arrow
+          fetch-depth: 0
+      - name: Store Release Tag Name as Environment Variable
+        # For convenience, store the release tag name as an environment
+        # variable for use in subsequent steps.
+        run: |
+          echo "ARROW_RELEASE_TAG_NAME=${{github.ref_name}}" >> "$GITHUB_ENV"
```

Comment:
We can refer `github.ref_name` as `${GITHUB_REF_NAME}`.

Reply:
Merged `publish_release.yml` and `publish_release_candidate.yml` into one file named `release.yml` and replaced all usage of `${{github.ref_name}}` with `${GITHUB_REF_NAME}` in `release.yml`. Marking as resolved.